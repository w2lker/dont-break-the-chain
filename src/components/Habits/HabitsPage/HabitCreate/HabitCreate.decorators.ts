import withStyles from '@material-ui/core/styles/withStyles';
import HabitCreate from './HabitCreate';
import habitCreateStyles from './HabitCreate.styles';

const styledComponent = withStyles(habitCreateStyles)(HabitCreate);

export default styledComponent;
