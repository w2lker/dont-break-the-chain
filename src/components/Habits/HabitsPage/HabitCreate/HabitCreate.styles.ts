const habitCreateStyles = {
  root: {
    padding: 0,
  },
};

export default habitCreateStyles;
