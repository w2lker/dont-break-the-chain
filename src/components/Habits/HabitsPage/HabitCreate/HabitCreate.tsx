import React from 'react';

import { WithStyles } from '@material-ui/styles';

import habitCreateStyles from './HabitCreate.styles';

export interface IHabitCreateProps extends WithStyles<typeof habitCreateStyles> {}

const HabitCreate: React.FC<IHabitCreateProps> = (props) => {
  const { classes } = props;
  return (
    <div className={classes.root}>
      HabitCreate Component
    </div>
  );
};

export default HabitCreate;
