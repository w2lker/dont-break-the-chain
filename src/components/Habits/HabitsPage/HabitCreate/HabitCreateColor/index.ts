import HabitCreateColor from './HabitCreateColor.decorators';

export default HabitCreateColor;
