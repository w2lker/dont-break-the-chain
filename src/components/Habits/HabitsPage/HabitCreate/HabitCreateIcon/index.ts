import HabitCreateIcon from './HabitCreateIcon.decorators';

export default HabitCreateIcon;
