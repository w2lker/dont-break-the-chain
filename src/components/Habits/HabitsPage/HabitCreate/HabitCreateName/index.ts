import HabitCreateName from './HabitCreateName.decorators';

export default HabitCreateName;
