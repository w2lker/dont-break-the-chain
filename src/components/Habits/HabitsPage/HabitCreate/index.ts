import HabitCreate from './HabitCreate.decorators';

export default HabitCreate;
