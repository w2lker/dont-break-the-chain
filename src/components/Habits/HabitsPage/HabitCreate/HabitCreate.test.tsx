import React from 'react';

import { mount, shallow } from 'enzyme';
// import cases from 'jest-in-case';

import HabitCreate, { IHabitCreateProps } from './HabitCreate';
import HabitCreateDecorated from './HabitCreate.decorators';
import habitCreateStyles from './HabitCreate.styles';

const testSetup = () => {
  const classes = {
    root: 'habitCreate-wrapper',
  };
  const classesKeys = Object.keys(classes);
  const emptyProps: IHabitCreateProps = {
    classes,
  };
  const sampleProps: IHabitCreateProps = {
    classes,
  };

  return { classes, classesKeys, emptyProps, sampleProps };
};

describe('HabitCreate component', () => {

  it('match snapshot', () => {
    const { sampleProps } = testSetup();
    const component = shallow(<HabitCreate {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('sampleTest', () => {
    expect(true).toBe(true);
  });

  test.todo('sample placeholder');
});

describe('HabitCreate styles', () => {
  it('contains principle classes', () => {
    const { classesKeys } = testSetup();
    expect(habitCreateStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(habitCreateStyles[keyValue]).toBeDefined();
    });
  });
  test.todo('sample placeholder');
});

describe('HabitCreate decorators', () => {
  it('provides styled classes from decorators', () => {
    const { sampleProps, classesKeys } = testSetup();
    const component = mount(<HabitCreateDecorated {...sampleProps} />);
    // @ts-ignore
    const assignedClasses = component.find('HabitCreate').props().classes;
    classesKeys.forEach(keyValue => {
      expect(assignedClasses[keyValue]).toBeDefined();
    });
  });
  test.todo('sample placeholder');
});
