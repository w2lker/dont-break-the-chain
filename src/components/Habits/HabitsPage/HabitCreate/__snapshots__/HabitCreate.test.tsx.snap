// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitCreate component match snapshot 1`] = `
"<div className=\\"habitCreate-wrapper\\">
  HabitCreate Component
</div>"
`;
